Resolve #743 - Add "author, updated ... ago and priority" in project page

KaiDalgleish opened the pull request:
**This resolves #743**  
Updated the project description to include more information about the project, based off of similar code in "home.mako".

![screen shot 2016-03-04 at 4 17 42 pm](https://cloud.githubusercontent.com/assets/11284580/13543909/afaa7f9c-e224-11e5-9570-9d845da392dd.png)
- [x] Author username appears (there is no listed author for the demo project, so it doesn't show up in the screenshot)
- [x] Last update appears
- [x] Priority appears as text, not a number

(This is my first pull request to this project, so please feel free to make suggestions and corrections)

pgiraud commented:
It looks very good!
Have you considered factorizing with `home.mako`? You copied some code from there, didn't you?
If this is too much work I can live with your change.

KaiDalgleish commented:
Good idea; I'll work on that today.

KaiDalgleish commented:
This was a little more complicated than I expected because in home.mako, the code was inside a function/def which is called for each project. Using `%include` to link to another template resulted in an AttributeError for any `project` reference.

I worked around that by making it into a separate `def` that could be imported and used anywhere. I put it in`custom

- [x] Made `display_project_info(project)` in `custom.mako`
    - Renders project status, author, update time, and priority with correct styling
- [x] Imported `display_project_info` into `home.mako`, `project.description.mako`


**Home View:**
![screen shot 2016-03-05 at 8 07 45 pm](https://cloud.githubusercontent.com/assets/11284580/13552206/08a4640c-e30e-11e5-8bc9-df246a779ecc.png)

**Project View:**
![screen shot 2016-03-05 at 8 08 01 pm](https://cloud.githubusercontent.com/assets/11284580/13552207/08a4ca14-e30e-11e5-9940-379b09d167d3.png)

KaiDalgleish commented:
Also, squashed commits :)

pgiraud commented:
It looks excellent. Creating a `def` is definitely the way to go.
I have one last request. The `custom.mako` file is intended to be used for elements that should be (or least can be) overridden by application administrators. I don't think that the "project info" is something that is supposed to be changed. It can remain the same all over the different instances.

I see two options:
 - creating a dedicated `project.info.mako` that you can include using `namespace`,
 - creating a new `helpers.mako` templates in which we can put everything template partial that can be included in several places.

pgiraud commented:
@KaiDalgleish Seen my last comment?

KaiDalgleish commented:
@pgiraud Nope, hadn't see your comment! Thanks for the feedback. I wondered if that was the right place for it. 
To fix this I ...
- [x] Made `helpers.mako` for `display_project_info`
- [x] Changed references in `home.mako` and `project.description.mako`
- [x] squashed commits

pgiraud commented on osmtm/templates/home.mako:
```diff
@@ -112,7 +113,7 @@ sorts = [('priority', 'asc', _('High priority first')),
     if request.locale_name:
         project.locale = request.locale_name
-    priority = priorities[project.priority]
+    # priority = priorities[project.priority]
```
Please don't leave any commented code. If it's not use just remove it.

pgiraud commented on osmtm/templates/project.description.mako:
```diff
@@ -16,13 +17,10 @@ from osmtm.mako_filters import markdown_filter
-</p>
-% endif
 <p>${project.description | markdown_filter, n}</p>
+<p id="info">
```
Any reason why you add an id for this element?

KaiDalgleish commented:
Thanks for feedback!
Cleaned up per @pgiraud 's comments:
- [x] Removed commented-out code
- [x] Removed unused `id`

pgiraud commented:
This looks great.
However, I just figured out that your pull request has been done against 2.12. Is there any reason for that? I think the target is rather master since it's not a bug fix.
If this OK for your, can you please create a new pull request?

KaiDalgleish commented:
Yeah, when I was first getting set up I had issues and was told to pull off of 2.12 instead of master. I'll submit a new request :)
